qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,229 @@
+        result_check = not_contain, expect_o should not be contained in qmp_o.
+
+        :param qmp_o: output from qmp_cmd.
+        :type qmp_o: list
+        :param expect_o: the expect result.
+        :type expect_o: list
+        """
+        logging.info("Expect result is %s" % expect_o)
+        logging.info("Actual result that get from qmp_cmd is %s" % qmp_o)
+        result = None
+        if result_check == "equal":
+            if not operator.eq(qmp_o, expect_o):
+                test.fail("QMP output does not equal to the expect result.\n "
+                          "Expect result: '%s'\n"
+                          "Actual result: '%s'" % (expect_o, qmp_o))
+        elif result_check == "contain":
+            for o in expect_o:
+                if isinstance(o, dict):
+                    for key, val in o.items():
+                        result = check_list(qmp_o, key, val)
+                        if not result:
+                            break
+                elif isinstance(o, str):
+                    result = check_list(qmp_o, o, check_item_in_pair=False)
+
+                if result:
+                    logging.info("QMP output contain the expect value %s" % o)
+                else:
+                    test.fail("QMP output does not contain the expect value.\n"
+                              "Missed expect value: '%s'\n"
+                              "Actual result: '%s'\n" % (o, qmp_o))
+
+    def execute_qmp_cmd(qmp_cmd, expect_result):
+        """
+        Execute qmp command and check if result as expect
+
+        :param qmp_cmd: qmp command
+        :type qmp_cmd: str
+        :param expect_result: expect result of qmp command
+        :type expect_result: str
+        """
+        # qmp command
+        try:
+            # Testing command
+            logging.info("Run qmp command '%s'.", qmp_cmd)
+            qmp_o = qmp_port.send_args_cmd(qmp_cmd)
+            if not isinstance(qmp_o, list):
+                qmp_o = [qmp_o]
+            logging.debug("QMP command: '%s' \n Output: '%s'", qmp_cmd, qmp_o)
+        except Exception as err:
+            qmp_o = err.data
+            logging.info(err)
+
+        if result_check == "equal" or result_check == "contain":
+            logging.info("Verify qmp command '%s'." % qmp_cmd)
+            check_result(qmp_o, eval(expect_result))
+
+    def check_dump_file(file_name):
+        """
+        Use crash to check dump file
+
+        :param file_name: dump file name
+        :type file_name: str
+
+        """
+        try:
+            kernel_version = process.getoutput("uname -r")
+            crash_script = "/home/crash.cmd"
+            process.run("echo bt > %s" % crash_script)
+            process.run("echo quit >> %s" % crash_script)
+            crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux "
+            crash_cmd %= (crash_script, kernel_version)
+            crash_cmd += file_name
+            output = process.getoutput(crash_cmd)
```

Comment:
Better check the output status, should be 0.

Reply:
Thank you for your suggestion. I modified it according to your suggestion, but I think `utils_package.package_install` only use to installs the package in the guest,  so I did not modify this.  I don't know more about this function,could you explain more about this function?
Many thanks!

Reply:
@leidwang `utils_package.package_install` works on host when you don't pass `session` to it, it finally goes to `software_manager.SoftwareManager.install()`. You might find several usages as example at avocado-vt/virttest if you search it there.

Reply:
Thank you very much!
Update the code.Could you please review this patch again?